File: fizzbuzz.py
def fizzbuzz(n):
    """Solves the FizzBuzz problem.

    >>> fizzbuzz(10)
    1
    2
    fizz
    4
    buzz
    6
    7
    8
    fizz
    buzz
    """
    for i in range(1, n + 1):
        if i % 3 == 0 and i % 5 == 0:
            print('fizzbuzz')
        elif i % 3 == 0:
            print('fizz')
        elif i % 5 == 0:
            print('buzz')
        else:
            print(i)

File: fizzbuzz.hs
module Fizzbuzz (
    fizzbuzz
) where

fizzbuzz :: [String]
fizzbuzz = map fbCond [1..]
    where
        fbCond n
            | n `mod` 3 == 0 && n `mod` 5 == 0 = "fizzbuzz"
            | n `mod` 3 == 0 = "fizz"
            | n `mod` 5 == 0 = "buzz"
            | otherwise = show n
